apps/www/pages/changelog.tsx

```diff
@@ -6,40 +6,219 @@ import CTABanner from '~/components/CTABanner'
 
-  if (!data) {
+export type Discussion = {
+  id: string
+  updatedAt: string
+  url: string
+  title: string
+  body: string
+}
+
+type Entry = {
+  id: string
+  title: string
+  url: string
+  created_at: string
+  source: MDXRemoteSerializeResult
+  type: string
+}
+
+export type DiscussionsResponse = {
+  repository: {
+    discussions: {
+      totalCount: number
+      nodes: Discussion[]
+      pageInfo: any
+    }
+  }
+}
+
+/**
+ * [Terry]
+ * this page powers supabase.com/changelog
+ * this page used to just be a feed of the releases endpoint
+ * (https://api.github.com/repos/supabase/supabase/releases) (rest api)
+ * but is now a blend of that legacy relases and the new Changelog category of the Discussions
+ * https://github.com/orgs/supabase/discussions/categories/changelog (graphql api)
+ * We should use the Changelog Discussions category for all future changelog entries and stop using releases
+ */
+
+export const getServerSideProps: GetServerSideProps = async ({ res, query }) => {
+  // refresh every 15 minutes
+  res.setHeader('Cache-Control', 'public, max-age=900, stale-while-revalidate=900')
+  const next = query.next ?? (null as string | null)
+  const restPage = query.restPage ? Number(query.restPage) : 1
+
+  const octokitRest = new OctokitRest({
+    auth: process.env.GITHUB_CHANGELOG_APP_REST_KEY,
+  })
+
+  // uses the rest api
+  async function fetchGitHubReleases() {
+    try {
+      const response = await octokitRest.repos.listReleases({
+        owner: 'supabase',
+        repo: 'supabase',
+        per_page: 10,
+        page: restPage,
+      })
+
+      return response.data || []
+    } catch (error) {
+      console.error(error)
+      return []
+    }
+  }
+
+  const releases = await fetchGitHubReleases()
+
+  // uses the graphql api
+  async function fetchDiscussions(owner: string, repo: string, categoryId: string, cursor: string) {
+    const octokit = new ExtendedOctokit({
+      authStrategy: createAppAuth,
+      auth: {
+        appId: process.env.GITHUB_CHANGELOG_APP_ID,
+        installationId: process.env.GITHUB_CHANGELOG_APP_INSTALLATION_ID,
+        privateKey: process.env.GITHUB_CHANGELOG_APP_PRIVATE_KEY,
+      },
+    })
+
+    const query = `
+      query troubleshootDiscussions($cursor: String, $owner: String!, $repo: String!, $categoryId: ID!) {
+        repository(owner: $owner, name: $repo) {
+          discussions(first: 10, after: $cursor, categoryId: $categoryId, orderBy: { field: CREATED_AT, direction: DESC }) {
+            totalCount
+            pageInfo {
+              hasPreviousPage
+              hasNextPage
+              startCursor
+              endCursor
+            }
+            nodes {
+              id
+              publishedAt
+              createdAt
+              url
+              title
+              # labels(first: 10) {
+                # nodes {
+                  # name
+                # }
+              # }
```

Comment:
Any reason this still needs to be here if it's commented out?

Reply:
aah right right was testing out a thing with that, will remove